scio-core/src/main/scala/com/spotify/scio/util/FilenamePolicyCreator.scala

```diff
@@ -0,0 +1,50 @@
+package com.spotify.scio.util
+
+import com.twitter.chill.ClosureCleaner
+import org.apache.beam.sdk.io.FileBasedSink
+import org.apache.beam.sdk.io.FileBasedSink.FilenamePolicy
+import org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions
+import org.apache.beam.sdk.io.fs.ResourceId
+import org.apache.beam.sdk.transforms.windowing.{BoundedWindow, PaneInfo}
+
+trait FilenamePolicyCreator {
+  def apply(path: String, suffix: String): FilenamePolicy
+}
+
+object FilenamePolicyCreator {
+  def filenamePolicyCreatorOf(
```

Comment:
I feel like the `Int, Int`s in the windowed and unwindowed signatures could be confusing for users because it's not immediately clear what they represent 🤔  I get not wanting to add more classes but even some syntactic sugar like this could help?

```scala
type ShardNum = Int
type NumShards = Int

def filenamePolicyCreatorOf(
    windowed: (ShardNum, NumShards, BoundedWindow, PaneInfo) => String = null,
    unwindowed: (ShardNum, NumShards) => String = null
  ): FilenamePolicyCreator = ...
```

Reply:
This is really only for use in the case of tests where brevity is prized. Otherwise users should just do `new FilenamePolicy` and make an impl, which is equivalent to what you're asking for, I think